pkg/libvmi/vm.go

```diff
@@ -62,6 +62,7 @@ func NewVirtualMachine(vmi *v1.VirtualMachineInstance, opts ...VMOption) *v1.Vir
 func WithRunning() VMOption {
 	return func(vm *v1.VirtualMachine) {
 		vm.Spec.Running = pointer.P(true)
+		vm.Spec.RunStrategy = nil
```

Comment:
@EdDev How do you feel about this? Do we want options to override the previous one?

Imo It can be hard to spot that a test was supposed to use `RunStrategy` but with mistake we used `Running`, consider example where we hide options behind some function with generic name.

Reply:
This has been discussed here.

This is not a good step, however due to its many usages we need a be pragmatic. I think that PR can take care of this and use an exception with a deprecation notice.

But the other builder (`WithRunStrategy`) is not good and I will not recommend it.

Reply:
Would you consider keeping it this as a temporary hack until remove https://github.com/kubevirt/kubevirt/issues/1199 ? I see it as a reasonable workaround in order to condense the problem in these two function, until it is gone and they can be cleaned up. Alternatively, we can wait for https://github.com/kubevirt/kubevirt/pull/12023 .

Reply:
> This has been discussed here.

I meant to link here the discussion that was done on this subject: https://github.com/kubevirt/kubevirt/pull/12023#discussion_r1660671345

> Alternatively, we can wait for https://github.com/kubevirt/kubevirt/pull/12023 .

I prefer you either wait for that PR or cherry-pick his [commit](https://github.com/kubevirt/kubevirt/pull/12023/commits/d461064da322e316816751ba5e73f08d115631fd) on this PR.

Reply:
Ohh there is another commit: https://github.com/kubevirt/kubevirt/pull/12023/commits/7b8c21561ec850c0276c9bd3903deac169c26ab6

Reply:
Note: I notice the `libvmi.NewVM...` defines `vm.Spec.Running = pointer.P(false)`, this seems wrong